### PR TITLE
feat: Serde support

### LINKED ISSUE #2: [Feat]: Serialize to/from Struct
We need to develop a system where you can serialize NBT data into a struct and vice versa.

## Possible solutions:
 - Utilize `serde`, however, the order of fields in the NBT tag (tag id, name, content) may result in significant boilerplate code.
 - Implement custom derive macro.

### PR DESCRIPTION
**Pull request by Norbiros**
Fixes #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,3 +65,29 @@ fn example(bytes: &mut Bytes) {
         .unwrap();
 }
 ```
+
+## Serde
+*Requires `serde` feature.*
+
+```ignore
```
Comment (SzczurekYT) [inline README.md:72]: I don't like this, because it breaks highligthing but haven't found any solution

File: src/serde/ser.rs [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        self,
+        _name: &'static str,
+        _variant_index: u32,
+        _variant: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeTupleVariant> {
+        Err(UnsupportedType("tuple variant".to_string()))
+    }
+
+    fn serialize_map(self, _len: Option<usize>) -> Result<Self::SerializeMap> {
```
Comment (SzczurekYT) [inline src/serde/ser.rs:282]: Map should be serialized just like a struct, as a compound.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -15,6 +15,6 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - name: Run Clippy
-        run: cargo clippy --all-targets --all-features
+        run: cargo clippy --all
       - name: Run tests
-        run: cargo test --verbose
\ No newline at end of file
+        run: cargo test --all --verbose
```
Comment (SzczurekYT) [inline .github/workflows/tests.yml:20]:
```suggestion
        run: cargo test --all-features --verbose
```

File: README.md (hunk 1 above)
Comment (Norbiros) [reply README.md:72]: What do you think about adding `doctest = false`?

Comment (SzczurekYT) [reply README.md:72]: If you can disabled just this one doctest while keeping the ```rust for proper highlighting then sure.

Comment (Norbiros) [reply README.md:72]: It disables testing all doctests

Comment (SzczurekYT) [reply README.md:72]: Wohoo, I managed to make it compile after your last force push

```diff
diff --git a/README.md b/README.md
index 76535c7..a7f3634 100644
--- a/README.md
+++ b/README.md
@@ -69,11 +69,11 @@ fn example(bytes: &mut Bytes) {
 ## Serde
 *Requires `serde` feature.*
 
-```ignore
-use crab_nbt::serde::arrays::IntArray;
+```rust
+use crab_nbt::serde::{arrays::IntArray, ser::to_bytes_unnamed, de::from_bytes_unnamed};
 use serde::{Deserialize, Serialize};
 
-#[derive(Deserialize, PartialEq, Debug)]
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
 struct Test {
     number: i32,
     #[serde(with = "IntArray")]
@@ -85,9 +85,9 @@ fn cycle() {
         number: 5,
         int_array: vec![7, 8]
     };
-    let bytes = to_bytes_unnamed(&test).unwrap();
-    let recreated_struct: Test = from_bytes_unnamed(&mut expected).unwrap();
+    let mut bytes = to_bytes_unnamed(&test).unwrap();
+    let recreated_struct: Test = from_bytes_unnamed(&mut bytes).unwrap();
     
     assert_eq!(test, recreated_struct);
 }
-```
\ No newline at end of file
+```
```

**Review by SzczurekYT — APPROVED** [review]

**Pull request merged**